src/BenchmarkDotNet/Validators/DotNetSdkVersionValidator.cs

```diff
@@ -0,0 +1,326 @@
+﻿using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Running;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace BenchmarkDotNet.Validators
+{
+    public static class DotNetSdkVersionValidator
+    {
+        private static readonly Lazy<List<string>> cachedFrameworkSdks =
+        new Lazy<List<string>>(() => GetInstalledFrameworkSdks().ToList(), true);
+
+        public static IEnumerable<ValidationError> ValidateCoreSdks(string? customDotNetCliPath, BenchmarkCase benchmark)
+        {
+            if (!HasRuntimeMoniker(benchmark))
+            {
+                return Enumerable.Empty<ValidationError>();
+            }
+
+            var runtimeMoniker = benchmark.Job.Environment.Runtime.RuntimeMoniker;
+            string requiredSdkVersion = GetSdkVersionFromMoniker(runtimeMoniker);
+
+            var installedSdks = GetInstalledDotNetSdks(customDotNetCliPath);
+
+            if (!installedSdks.Any(sdk => sdk.StartsWith(requiredSdkVersion)))
+            {
+                return new ValidationError[]
+                {
+                    new ValidationError(true, $"The required .NET Core SDK version {requiredSdkVersion} for runtime moniker {runtimeMoniker} is not installed.", benchmark)
+                };
+            }
+
+            return Enumerable.Empty<ValidationError>();
+        }
+
+        public static IEnumerable<ValidationError> ValidateFrameworkSdks(BenchmarkCase benchmark)
```

Comment:
This could also use the `IsCliPathInvalid` check since the `CsProjClassicNetToolchain` uses it. Or it could be a separate helper function moved to this class and the toolchain can call both.